src/containers/pipelines/index.js

```diff
@@ -7,33 +7,41 @@ import { getPipelines as getPipelinesAction } from 'actions/pipelines';
 import { StyledTitle } from 'styles/app';
 import { StyledPipelines, StyledPipelinesButton } from 'styles/pipelines';
 import PipelineItem from './pipeline';
+import AddModal from './AddModal';
 
-class Pipelines extends Component {
-  componentDidMount() {
-    const { getPipelines } = this.props;
+function Pipelines({ getPipelines, pipelines }) {
+  useEffect(() => {
     getPipelines();
-  }
-
-  render() {
-    const { pipelines } = this.props;
-    return (
-      <>
-        <StyledTitle>
-          Pipelines
-          <StyledPipelinesButton type="text">
-            + Add Pipeline
-          </StyledPipelinesButton>
-        </StyledTitle>
-        <StyledPipelines>
-          {
-            pipelines.map(({ id, name, last_updated: lastUpdated }) => (
-              <PipelineItem key={id} name={name} lastUpdated={lastUpdated} />
-            ))
-          }
-        </StyledPipelines>
-      </>
-    );
-  }
+  }, []);
+
+  const [addModalVisible, setAddModalVisible] = useState(false);
+
+  const openModal = () => setAddModalVisible(true);
```

Comment:
could we get away with `toggleModal` ?
Unless we need to take specific actions during open/close